src/lightning/app/cli/commands/cp.py

```diff
@@ -0,0 +1,229 @@
+import concurrent
+import os
+import sys
+from functools import partial
+from multiprocessing.pool import ApplyResult
+from pathlib import Path
+from time import sleep
+from typing import Optional, Tuple
+
+import click
+import requests
+import rich
+import urllib3
+from lightning_cloud.openapi import IdArtifactsBody
+from rich.live import Live
+from rich.progress import BarColumn, DownloadColumn, Progress, Task, TextColumn
+from rich.spinner import Spinner
+from rich.text import Text
+
+from lightning.app.cli.commands.pwd import _pwd
+from lightning.app.source_code import FileUploader
+from lightning.app.utilities.app_helpers import Logger
+from lightning.app.utilities.network import LightningClient
+
+logger = Logger(__name__)
+
+
+@click.argument("src_path", required=True)
+@click.argument("dst_path", required=True)
+@click.option("-r", required=False, hidden=True)
+@click.option("--recursive", required=False, hidden=True)
+def cp(src_path: str, dst_path: str, r: bool = False, recursive: bool = False) -> None:
+    """Copy files between your local filesystem and the Lightning Cloud filesystem."""
+
+    if sys.platform == "win32":
+        print("`cp` isn't supported on windows. Open an issue on Github.")
+        sys.exit(0)
+
+    with Live(Spinner("point", text=Text("pending...", style="white")), transient=True) as live:
+
+        pwd = _pwd()
+
+        if pwd == "/" or len(pwd.split("/")) == 1:
+            return _error_and_exit("Uploading files at the project level isn't supported yet.")
+
+        client = LightningClient()
```

Comment:
Should `retry` be turned off? because otherwise it might keep sending requests from the command line execution silently:
```suggestion
        client = LightningClient(retry=False)
```